### PR TITLE
Base Sepolia Granite HF prestate upgrade

### PR DESCRIPTION
**Pull request by 0x00101010**
**Description**

Base Sepolia Granif HF prestate upgrade

**Tests**

https://dashboard.tenderly.co/TENDERLY_USERNAME/TENDERLY_PROJECT/simulator/new?network=11155111&contractAddress=0x0fe884546476dDd290eC46318785046ef68a0BA9&from=0x4672425C27A942bB27e7b9709c1b21ab89a3cA13&stateOverrides=%5B%7B"contractAddress":"0x0fe884546476dDd290eC46318785046ef68a0BA9","storage":%5B%7B"key":"0x0000000000000000000000000000000000000000000000000000000000000004","value":"0x0000000000000000000000000000000000000000000000000000000000000001"%7D%5D%7D,%7B"contractAddress":"0x0000000000000000000000000000000000000000","storage":%5B%5D%7D%5D&rawFunctionInput=0x6a761202000000000000000000000000ca11bde05977b3631167028862be2a173976ca110000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000014000000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000003c0000000000000000000000000000000000000000000000000000000000000024482ad56cb0000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000400000000000000000000000000000000000000000000000000000000000000120000000000000000000000000d6e6dbf4f7ea0ac412fd8b65ed297e64bb7a06e100000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000060000000000000000000000000000000000000000000000000000000000000004414f6b1a3000000000000000000000000000000000000000000000000000000000000000000000000000000000000000048f9f3190b7b5231cbf2ad1a1315af7f6a55402000000000000000000000000000000000000000000000000000000000000000000000000000000000d6e6dbf4f7ea0ac412fd8b65ed297e64bb7a06e100000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000060000000000000000000000000000000000000000000000000000000000000004414f6b1a3000000000000000000000000000000000000000000000000000000000000000100000000000000000000000054966d5a42a812d0daade1fa2321ff8b102d1ee1000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000410000000000000000000000004672425c27a942bb27e7b9709c1b21ab89a3ca1300000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000000

### CODE REVIEW

File: tasks/sep/base-003-fp-granite-prestate/.env [hunk 1]
```diff
@@ -0,0 +1,6 @@
+ETH_RPC_URL="https://gateway.tenderly.co/public/sepolia"
+SUPERCHAIN_CONFIG_ADDR=0xC2Be75506d5724086DEB7245bd260Cc9753911Be
+OWNER_SAFE=0x0fe884546476dDd290eC46318785046ef68a0BA9
```
Comment (0x00101010) [inline tasks/sep/base-003-fp-granite-prestate/.env:3]: https://sepolia.etherscan.io/address/0xd6E6dBf4F7EA0ac412fD8b65ED297e64BB7a06E1#readProxyContract#F8

Comment (0x00101010) [reply tasks/sep/base-003-fp-granite-prestate/.env:3]: DisputeGameProxy: `0xd6E6dBf4F7EA0ac412fD8b65ED297e64BB7a06E1`

File: tasks/sep/base-003-fp-granite-prestate/README.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+## State Validation
+
+Please see the instructions for [validation](./VALIDATION.md).
+
+## Execution
+
+This upgrade changes the `ABSOLUTE_PRESTATE` in the `FaultDisputeGame` and `PermissionedDisputeGame` to that of the `op-program/v1.3.0-rc.2` tag in preparation for the Granite hardfork.
+
+The batch will be executed on chain ID `11155111`, and contains `2` transactions.
```
Comment (0x00101010) [inline tasks/sep/base-003-fp-granite-prestate/README.md:40]: tx happens on sepolia L1

File: tasks/sep/base-003-fp-granite-prestate/README.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+## Tx #1: Upgrade `FaultDisputeGame`
+
+Upgrades the implementation of the `FaultDisputeGame` in the `DisputeGameFactory` to contain the updated absolute prestate value.
+
+**Function Signature:** `setImplementation(uint32,address)`
+
+**To:** `0xd6E6dBf4F7EA0ac412fD8b65ED297e64BB7a06E1`
```
Comment (0x00101010) [inline tasks/sep/base-003-fp-granite-prestate/README.md:48]: Verify against: https://docs.base.org/docs/base-contracts/: `DisputeGameFactoryProxy` and / or https://github.com/ethereum-optimism/superchain-registry/blob/main/superchain/configs/sepolia/base.toml#L50

File: tasks/sep/base-003-fp-granite-prestate/README.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+**Value:** `0 WEI`
+
+**Raw Input Data:** `0x14f6b1a30000000000000000000000000000000000000000000000000000000000000000000000000000000000000000da35a941d76ff37a4abd877156f58e6aa2e499b0`
```
Comment (0x00101010) [inline tasks/sep/base-003-fp-granite-prestate/README.md:52]: Check tenderly to confirm state transition

File: tasks/sep/base-003-fp-granite-prestate/README.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+### Inputs
+
+**\_gameType:** `0`
+
+**\_impl:** `0xDa35a941D76ff37A4abd877156F58E6Aa2e499b0`
```
Comment (0x00101010) [inline tasks/sep/base-003-fp-granite-prestate/README.md:58]: _gameType: 0 => FaultDisputeGame => `0xDa35a941D76ff37A4abd877156F58E6Aa2e499b0` (new implementation) => verify against above `SignFromJson.s.sol` `faultDisputeGame` var

File: tasks/sep/base-003-fp-granite-prestate/README.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+## Tx #2: Upgrade `PermissionedDisputeGame`
+
+Upgrades the implementation of the `PermissionedDisputeGame` in the `DisputeGameFactory` to contain the updated absolute prestate value.
+
+**Function Signature:** `setImplementation(uint32,address)`
+
+**To:** `0xd6E6dBf4F7EA0ac412fD8b65ED297e64BB7a06E1`
```
Comment (0x00101010) [inline tasks/sep/base-003-fp-granite-prestate/README.md:66]: Verify against: https://docs.base.org/docs/base-contracts/: DisputeGameFactoryProxy and / or https://github.com/ethereum-optimism/superchain-registry/blob/main/superchain/configs/sepolia/base.toml#L50

File: tasks/sep/base-003-fp-granite-prestate/README.md [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+**Value:** `0 WEI`
+
+**Raw Input Data:** `0x14f6b1a300000000000000000000000000000000000000000000000000000000000000010000000000000000000000008105d699b38ed7cf9b75353baccca60496ed72fd`
+
+### Inputs
+
+**\_gameType:** `1`
+
+**\_impl:** `0x8105D699B38eD7CF9b75353BaCccA60496Ed72Fd`
```
Comment (0x00101010) [inline tasks/sep/base-003-fp-granite-prestate/README.md:76]: Verify against `input.json` and tenderly to confirm state transition

File: tasks/sep/base-003-fp-granite-prestate/SignFromJson.s.sol [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+        0xDa3037Ff70Ac92CD867c683BD807e5A484857405; // In registry addresses.
+    address constant systemConfigOwner =
+        0x0fe884546476dDd290eC46318785046ef68a0BA9; // In registry addresses.
+    address constant batchSenderAddress =
+        0x6CDEbe940BC0F26850285cacA097C11c33103E47; // In registry genesis-system-configs
+    address constant p2pSequencerAddress =
+        0xb830b99c95Ea32300039624Cb567d324D4b1D83C; // cast call $SystemConfig "unsafeBlockSigner()(address)"
+    address constant batchInboxAddress =
+        0xfF00000000000000000000000000000000084532; // In registry yaml.
+    address constant currentPDGProposer =
```
Comment (0x00101010) [inline tasks/sep/base-003-fp-granite-prestate/SignFromJson.s.sol:39]: This is the current permissioned dispute game proposer address: https://sepolia.etherscan.io/address/0x3f5c770f17A6982d2B3Ac77F6fDC93BFE0330E17#readContract#F24

File: tasks/sep/base-003-fp-granite-prestate/SignFromJson.s.sol [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+        0x727D7c7fCa14b7F3C49a1C816b42a41fe2F709F9;
+    address constant currentPDGChallenger =
```
Comment (0x00101010) [inline tasks/sep/base-003-fp-granite-prestate/SignFromJson.s.sol:41]: https://sepolia.etherscan.io/address/0x3f5c770f17A6982d2B3Ac77F6fDC93BFE0330E17#readContract#F3

File: tasks/sep/base-003-fp-granite-prestate/SignFromJson.s.sol [hunk 10]
```diff
@@ -0,0 +1,199 @@
...
+        );
+        require(
+            address(permissionedDisputeGame) ==
+                address(dgfProxy.gameImpls(GameTypes.PERMISSIONED_CANNON)),
+            "check-100"
+        );
+        require(
+            faultDisputeGame.absolutePrestate().raw() ==
+                bytes32(
+                    0x0385c3f8ee78491001d92b90b07d0cf387b7b52ab9b83b4d87c994e92cf823ba
```
Comment (0x00101010) [inline tasks/sep/base-003-fp-granite-prestate/SignFromJson.s.sol:189]: New prestate, same as op's: https://github.com/ethereum-optimism/superchain-ops/pull/292/files#diff-a2e152dd79afc88135859112752bff6cc081bfbdaf1493a0b855c48d18860d84R147

File: tasks/sep/base-003-fp-granite-prestate/input.json [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+    },
+    "transactions": [
+        {
+            "metadata": {
+                "name": "Upgrade `FaultDisputeGame`",
+                "description": "Upgrades the implementation of the `FaultDisputeGame` in the `DisputeGameFactory` to a new implementation that contains the updated absolute prestate value."
+            },
+            "to": "0xd6E6dBf4F7EA0ac412fD8b65ED297e64BB7a06E1",
+            "value": "0x0",
+            "data": "0x14f6b1a30000000000000000000000000000000000000000000000000000000000000000000000000000000000000000da35a941d76ff37a4abd877156f58e6aa2e499b0",
```
Comment (0x00101010) [inline tasks/sep/base-003-fp-granite-prestate/input.json:15]: Verify against state transition in tenderly

File: tasks/sep/base-003-fp-granite-prestate/input.json [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+                        "name": "_impl",
+                        "type": "address"
+                    }
+                ],
+                "outputs": [],
+                "stateMutability": "nonpayable"
+            },
+            "contractInputsValues": {
+                "_impl": "0xDa35a941D76ff37A4abd877156F58E6Aa2e499b0",
+                "_gameType": "0"
```
Comment (0x00101010) [inline tasks/sep/base-003-fp-granite-prestate/input.json:34]: FaultDisputeGame

File: tasks/sep/base-003-fp-granite-prestate/input.json [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+                        "name": "_impl",
+                        "type": "address"
+                    }
+                ],
+                "outputs": [],
+                "stateMutability": "nonpayable"
+            },
+            "contractInputsValues": {
+                "_impl": "0x8105D699B38eD7CF9b75353BaCccA60496Ed72Fd",
+                "_gameType": "1"
```
Comment (0x00101010) [inline tasks/sep/base-003-fp-granite-prestate/input.json:63]: PermissionedDisputeGame

**Comment by 0x00101010** [comment]
> HI @0x00101010 , just to clarify, anybody from https://github.com/orgs/ethereum-optimism/teams/coinbase should be able to approve and merge this PR, so by default we will rely on https://github.com/orgs/ethereum-optimism/teams/coinbase's review since that group has the most context, but if you would like anybody else to review, feel free to put them in the `Assignee` field to signify that. Thank you!

Thanks @zchn, the link you pasted goes to 404 for me, not sure if it's permission or just wrong link.

File: tasks/sep/base-003-fp-granite-prestate/.env (hunk 1 above)
Comment (awilliams1-cb) [reply tasks/sep/base-003-fp-granite-prestate/.env:3]: Confirmed [Owner safe](https://github.com/ethereum-optimism/superchain-registry/blob/main/superchain/configs/sepolia/base.toml#L34) and [DisputeGameFactoryProxy](https://github.com/ethereum-optimism/superchain-registry/blob/main/superchain/configs/sepolia/base.toml#L50) with superchain-registry as well.

File: tasks/sep/base-003-fp-granite-prestate/SignFromJson.s.sol [hunk 14]
```diff
@@ -0,0 +1,198 @@
...
+        0x727D7c7fCa14b7F3C49a1C816b42a41fe2F709F9;
+    address constant currentPDGChallenger =
+        0x8b8c52B04A38f10515C52670fcb23f3C4C44474F;
+
+    // Currenet dispute game implementations
+    FaultDisputeGame currentFDG;
+    PermissionedDisputeGame currentPDG;
+
+    // New dispute game implementations
+    // https://sepolia.etherscan.io/address/0x48F9F3190b7B5231cBf2aD1A1315AF7f6A554020#code
```
Comment (0x00101010) [inline tasks/sep/base-003-fp-granite-prestate/SignFromJson.s.sol:48]: New implementation prestate points to the correct one: https://sepolia.etherscan.io/address/0x48F9F3190b7B5231cBf2aD1A1315AF7f6A554020#readContract#F1

File: tasks/sep/base-003-fp-granite-prestate/SignFromJson.s.sol [hunk 15]
```diff
@@ -0,0 +1,198 @@
...
+    FaultDisputeGame constant faultDisputeGame =
+        FaultDisputeGame(0x48F9F3190b7B5231cBf2aD1A1315AF7f6A554020);
+    // https://sepolia.etherscan.io/address/0x54966d5A42a812D0dAaDe1FA2321FF8b102d1ee1#code
```
Comment (0x00101010) [inline tasks/sep/base-003-fp-granite-prestate/SignFromJson.s.sol:51]: New implementation prestate points to the correct one: https://sepolia.etherscan.io/address/0x54966d5A42a812D0dAaDe1FA2321FF8b102d1ee1#readContract#F1

File: tasks/sep/base-003-fp-granite-prestate/README.md (hunk 2 above)
Comment (awilliams1-cb) [reply tasks/sep/base-003-fp-granite-prestate/README.md:40]: confirmed

File: tasks/sep/base-003-fp-granite-prestate/README.md (hunk 3 above)
Comment (awilliams1-cb) [reply tasks/sep/base-003-fp-granite-prestate/README.md:48]: `DisputeGameFactoryProxy` address confirmed

File: tasks/sep/base-003-fp-granite-prestate/SignFromJson.s.sol (hunk 8 above)
Comment (0x00101010) [reply tasks/sep/base-003-fp-granite-prestate/SignFromJson.s.sol:38]: Needs to be changed

File: tasks/sep/base-003-fp-granite-prestate/SignFromJson.s.sol (hunk 9 above)
Comment (0x00101010) [reply tasks/sep/base-003-fp-granite-prestate/SignFromJson.s.sol:40]: needs to be audited

**Review by anikaraghu — APPROVED** [review]

**Review by awilliams1-cb — DISMISSED** [review]

**Review by anikaraghu — APPROVED** [review]

**Pull request closed without merging**